design/open/gc-on-parquet.md

```diff
@@ -0,0 +1,25 @@
+## Garbage collection using Parquet format
+
+### Motivation
+
+Garbage collection in lakeFS is struggling at large scale.
+
+### Why Parquet?
+
+Parquet is more than a file format. It's a breathing open-source project aiming to provide exactly what we're looking for here: efficient data retrieval. GC is essentially a huge anti-join, and it may be very difficult to achieve a similar optimization in our dedicated format.
+
+
+### Steps
+
+#### 1. Sync
+Ranges of the repository are translated into Parquet files. These are stored in a new metadata location under the repository's storage namespace.
+
+#### 2. Sanity
+Verify all the ranges required for this run are present, and that the number of entries in each range is correct.
```

Comment:
Not sure why this is needed.  If it's a parquet from a range, it's good.  (If we don't trust it, why trust anything else, really?)

Reply:
At the very least, I want to validate that all the ranges are _there_.
Missing a range is a disaster. 
Since we're creating it using our InputFormat, I want to double-check that we have the correct number of addresses. Again, missing an address is also a disaster.